core/modules/replica/Configuration.cc

```diff
@@ -0,0 +1,343 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/Configuration.h"
+
+// System headers
+#include <stdexcept>
+
+// Qserv headers
+#include "lsst/log/Log.h"
+#include "replica/ConfigurationFile.h"
+#include "replica/ConfigurationMySQL.h"
+#include "replica/FileUtils.h"
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.Configuration");
+
+/**
+ * Print the content of a vector into the provided stream.
+ * Assume that proper astreaming operator is defined for the vector
+ * elements.
+ *
+ * @param os - the output stream object
+ * @param v  - the vector to stream
+ * @return   - the stream object
+ */
+template <typename T>
+void vector2stream(std::ostream&         os,
+                   std::vector<T> const& v) {
+    for (size_t i = 0, num = v.size(); i < num; ++i) {
+        os  << (i ? "," : "") << v[i];
+    }
+}
+
+} // namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+std::ostream& operator <<(std::ostream& os, WorkerInfo const& info) {
+    os  << "WorkerInfo ("
+        << "name:'"      <<      info.name       << "',"
+        << "isEnabled:"  << (int)info.isEnabled  << ","
+        << "isReadOnly:" << (int)info.isReadOnly << ","
+        << "svcHost:'"   <<      info.svcHost    << "',"
+        << "svcPort:"    <<      info.svcPort    << ","
+        << "fsHost:'"    <<      info.fsHost     << "',"
+        << "fsPort:"     <<      info.fsPort     << ","
+        << "dataDir:'"   <<      info.dataDir    << "')";
+    return os;
+}
+
+std::ostream& operator <<(std::ostream& os, DatabaseInfo const& info) {
+    os  << "DatabaseInfo ("
+        << "name:'" << info.name << "',"
+        << "family:'" << info.family << "',"
+        << "partitionedTables:[";
+    ::vector2stream(os, info.partitionedTables);
+    os  << "],"
+        << "regularTables:[";
+    ::vector2stream(os, info.regularTables);
+    os  << "])";
+    return os;
+}
+
+Configuration::pointer Configuration::load(std::string const& configUrl) {
+
+    for (auto const& proposedPrefix: std::vector<std::string>{"file:","mysql:"}) {

```

Comment:
This simplification will trigger a chain reaction of modification within the loop because I will have to deal with plain vanilla C-style string instead of std::string. That was a motivation for this way of construction.

Reply:
Rewritten